node/lib/cmd/submodule.js

```diff
@@ -0,0 +1,136 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+"use strict";
+
+const co = require("co");
+
+/**
+ * This module contains methods for pulling.
+ */
+
+/**
+ * help text for the `pull` command
+ * @property {String}
+ */
+exports.helpText = `Submodule-specific commands.`;
+
+/**
+ * description of the `pull` command
+ * @property {String}
+ */
+exports.description =`
+Provide commands pertaining to submodules that are not provided, easily or
+efficiently by 'git submodule'.`;
+
+exports.configureParser = function (parser) {
+
+    const subParsers = parser.addSubparsers({
+        dest: "command",
+    });
+
+    const statusParser = subParsers.addParser("status", {
+        help: "show information about submodules",
+        description: `
+The default behavior is to show a one-line summary of each open submodule: the
+current SHA-1 for that submodule followed by its name.`,
+    });
+
+    statusParser.addArgument(["path"], {
+        type: "string",
+        help: "show information about only the submodules in these paths",
+        nargs: "*",
+    });
+
+    statusParser.addArgument(["-v", "--verbose"], {
+        defaultValue: false,
+        required: false,
+        action: "storeConst",
+        constant: true,
+        help: `show one-line summary of each submodule: a '-' if the \
+submodule is closed, followed by the current SHA-1 for \
+that submodule, followed by its name. `
+    });
+};
+
+/**
+ * Execute the pull command according to the specified `args`.
+ *
+ * @async
+ * @param {Object}  args
+ * @param {Boolean} args.any
+ * @param {String}  repository
+ * @param {String}  [source]
+ */
+exports.executeableSubcommand = co.wrap(function *(args) {
+    // TODO: right now, there is only one sub-command, "status".  Later we will
+    // have to split this out.
+
+    const colors = require("colors");
+    const path   = require("path");
+
+    const GitUtil       = require("../util/git_util");
+    const Status        = require("../util/status");
+
+    const repo = yield GitUtil.getCurrentRepo();
+    const workdir = repo.workdir();
+    const cwd = process.cwd();
+    const paths = yield args.path.map(filename => {
+        return GitUtil.resolveRelativePath(workdir, cwd, filename);
+    });
+    const repoStatus = yield Status.getRepoStatus(repo, {
+        paths: paths,
+        showMetaChanges: false,
```

Comment:
This flag controls whether or not changes to the meta-repo workdir and index are loaded. Changes to the meta-repo workdir and index cannot affect the output of this command.